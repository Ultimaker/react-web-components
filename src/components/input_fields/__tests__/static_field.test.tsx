// Copyright (c) 2018 Ultimaker B.V.
import * as React from 'react';
import { shallow } from 'enzyme';

// components
import StaticField, {StaticFieldProps} from '../static_field';
import InputFieldWrapper from '../input_field_wrapper';

describe('The static field component', () => {
    let props: StaticFieldProps;
    let wrapper;

    beforeEach(() => {
        props = {
            id: 'testInputField',
            value: "2018",
            labelLayoutWidth: '1/1',
            labelWidthBreakpoint: 'sm',
        };
        wrapper = shallow(<StaticField {...props}>child</StaticField>);
    });

    it('should render an input wrapper', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it("should render it's value and its children", () => {
        expect(wrapper.children().text()).toEqual("2018");
        expect(wrapper.find(InputFieldWrapper).prop('inputChildren')).toEqual("child");
    });

    it('should render a null', () => {
        wrapper.setProps({value: null})
        expect(wrapper.find(InputFieldWrapper).prop("value")).toBeNull();
    })
});
