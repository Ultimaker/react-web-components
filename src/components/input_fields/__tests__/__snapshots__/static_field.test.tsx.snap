// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The static field component should render an input wrapper 1`] = `
<InputFieldWrapper
  id="testInputField"
  inputChildren="child"
  labelLayoutWidth="1/1"
  labelWidthBreakpoint="sm"
  staticField={true}
  touched={false}
  value="2018"
>
  2018
  child
</InputFieldWrapper>
`;
